packages/core/test/requestCollection.spec.ts

```diff
@@ -1,350 +1,153 @@
+import { FetchProxy } from '../src/fetchProxy'
 import { Observable } from '../src/observable'
 import {
-  isRejected,
-  isServerError,
   RequestCompleteEvent,
   RequestObservables,
   RequestStartEvent,
   RequestType,
   trackFetch,
   trackXhr,
 } from '../src/requestCollection'
-import { FetchStub, FetchStubBuilder, FetchStubPromise, isIE } from '../src/specHelper'
-import { find, includes } from '../src/utils'
+import { FetchStub, FetchStubManager, isIE, stubFetch, withXhr } from '../src/specHelper'
+import { XhrProxy } from '../src/xhrProxy'
 
-describe('fetch tracker', () => {
+describe('collect fetch', () => {
   const FAKE_URL = 'http://fake-url/'
-  let originalFetch: any
-  let fetchStubBuilder: FetchStubBuilder
-  let fetchStub: (input: RequestInfo, init?: RequestInit) => FetchStubPromise
+  let fetchStub: FetchStub
+  let fetchStubManager: FetchStubManager
+  let fetchProxy: FetchProxy
+  let startSpy: jasmine.Spy
+  let completeSpy: jasmine.Spy
```

Comment:
this could be properly typed, so we wouldn't need as much casts
```suggestion
  let startSpy: jasmine.Spy<(requestStartEvent: RequestStartEvent) => void>
  let completeSpy: jasmine.Spy<(requestCompleteEvent: RequestCompleteEvent) => void>
```